### PR TITLE
OAK-10611: o.a.j.o.namepath.JcrPathParser does not handle invalid characters correctly

### PR DESCRIPTION
**Pull request by mbaedke**
…racters correctly

Fixed treatment of square brackets in some edge cases.

### CODE REVIEW

File: oak-core-spi/src/test/java/org/apache/jackrabbit/oak/namepath/PathParserTest.java [hunk 1]
```diff
@@ -301,20 +300,21 @@ public void testUnexpectedOpeningSquareBracket() throws RepositoryException {
...
     @Test
-    @Ignore  //OAK-10611
     public void testMissingClosingSquareBracket() throws RepositoryException {
         String path = "/a[";
         TestListener listener = new TestListener(
                 CALLBACKRESULT_ROOT,
+                CALLBACKRESULT_ERROR_ANY
                 //the parser actually produces an error, but we should change the error message to something like this
-                CALLBACKRESULT_ERROR(errorClosingQuareBracketExpected(path))
+                //CALLBACKRESULT_ERROR(errorClosingQuareBracketExpected(path))
```
Comment (reschke) [inline oak-core-spi/src/test/java/org/apache/jackrabbit/oak/namepath/PathParserTest.java]: Should these comments be removed?

Comment (mbaedke) [reply oak-core-spi/src/test/java/org/apache/jackrabbit/oak/namepath/PathParserTest.java]: I cleaned up the comments

**Review by reschke — APPROVED** [review]
